packages/components/psammead-topic-tags/src/index.test.jsx

```diff
@@ -0,0 +1,79 @@
+import React from 'react';
+import { shouldMatchSnapshot } from '@bbc/psammead-test-helpers';
+import { latin, burmese, arabic } from '@bbc/gel-foundations/dist/scripts';
+import { TopicTags, TopicTag } from './index';
+
+describe('TopicTags', () => {
+  const baseProps = {
+    dir: 'ltr',
+    service: 'news',
+    script: latin,
+  };
+
+  shouldMatchSnapshot(
+    'should correctly render a single topic for ltr',
+    <TopicTags {...baseProps}>
+      <TopicTag topicName="test1" topicLink="#" />
+    </TopicTags>,
+  );
+
+  shouldMatchSnapshot(
+    'should correctly render a single topic for rtl',
+    <TopicTags {...baseProps} dir="rtl">
+      <TopicTag topicName="test1" topicLink="#" />
+    </TopicTags>,
+  );
+
+  shouldMatchSnapshot(
+    'should correctly render multiple topic tags for ltr',
+    <TopicTags {...baseProps}>
+      <TopicTag topicName="test1" topicLink="#" />
+      <TopicTag topicName="test2" topicLink="#" />
+      <TopicTag topicName="test3" topicLink="#" />
+      <TopicTag topicName="test4" topicLink="#" />
+    </TopicTags>,
+  );
+
+  shouldMatchSnapshot(
+    'should correctly render multiple topic tags for rtl',
+    <TopicTags {...baseProps} dir="rtl">
+      <TopicTag topicName="test1" topicLink="#" />
+      <TopicTag topicName="test2" topicLink="#" />
+      <TopicTag topicName="test3" topicLink="#" />
+      <TopicTag topicName="test4" topicLink="#" />
+    </TopicTags>,
+  );
+
+  shouldMatchSnapshot(
+    'should correctly render a single topic tag for burmese',
+    <TopicTags service="burmese" script={burmese} dir="rtl">
+      <TopicTag topicName="test1" topicLink="#" />
+    </TopicTags>,
+  );
+
+  shouldMatchSnapshot(
+    'should correctly render multiple topic tags for arabic',
+    <TopicTags service="arabic" script={arabic} dir="rtl">
+      <TopicTag topicName="test1" topicLink="#" />
+      <TopicTag topicName="test2" topicLink="#" />
+      <TopicTag topicName="test3" topicLink="#" />
+      <TopicTag topicName="test4" topicLink="#" />
+    </TopicTags>,
+  );
+
+  shouldMatchSnapshot(
+    'should not render if there are no children',
+    <TopicTags {...baseProps} />,
+  );
+
+  shouldMatchSnapshot(
+    'should ignore non-TopicTag children',
+    <TopicTags {...baseProps}>
+      <TopicTag topicName="test1" topicLink="#" />
+      <div>
+        <p>ignore</p>
+      </div>
+      <TopicTag topicName="test2" topicLink="#" />
```

Comment:
For these tests that essentially yield an empty snapshot, I think it may be a bit more readable to render the component and assert that it has rendered nothing. I tend to use snapshots where you want to capture a complex nested structure of DOM elements as a snapshot; to assert the full structure would entail many assertions. 

Where you want test a single specific characteristic, e.g. a specific attribute is added like `dir` or in this case we are just rendering nothing, I tend to favour assetions for readability as they more clearly specify the what you are expecting.

Reply:
Thanks @andrewscfc,

I've rewritten these 2 tests which check for invalid usage, using the `expect` method. Is this ok? I haven't much experience with Jest, but they do seem to work. 

https://github.com/bbc/psammead/blob/a067c3767c07e9ea9c041c9e9d06ab95db1ddecf/packages/components/psammead-topic-tags/src/index.test.jsx#L65-L85

Reply:
Yeah looks good, a bit more clear what you are testing for imo. Thanks 👍